spec/rubocop/cop/lint/useless_access_modifier_spec.rb

```diff
@@ -291,6 +291,28 @@
     end
   end
 
+  context 'when using ActiveSupport behavior when Rails is not eabled' do
+    it 'reports offenses' do
+      src = ['module SomeModule',
+             '  extend ActiveSupport::Concern',
+             '  class_methods do',
+             '    def some_public_class_method',
+             '    end',
+             '    private',
+             '    def some_private_class_method',
+             '    end',
+             '  end',
+             '  def some_public_instance_method',
+             '  end',
+             '  private',
+             '  def some_private_instance_method',
+             '  end',
+             'end']
+      inspect_source(cop, src)
+      expect(cop.offenses.size).to eq(1)
```

Comment:
I think maybe it should, although it's a kind of awkward scenario to imagine. If someone is using a `class_methods` method which takes a block, but hasn't opted into Rails-related checks, RuboCop *arguably* should make no assumptions about the behavior of the `class_methods` method.

The purpose of this PR is to suppress false positives when using a non-Rails cop in a Rails (or maybe just an ActiveSupport) context. Maybe @bbatsov or @alexdowad will prefer not to introduce more Rails awareness into non-Rails cops, because it will make it harder to extract those into a separate gem (a plan I've seen them mention a few times before).  I'm not sure how else to avoid these false positives, though, and there is [some precedent](https://github.com/bbatsov/rubocop/blob/591b7becabb9fd7ad4ddc06ad2d396e884c849fe/lib/rubocop/cop/performance/count.rb#L79-L83) for this kind of thing.